pkg/nodeconfig/nodeconfig.go

```diff
@@ -255,25 +249,27 @@ func (nc *nodeConfig) configureNetwork() error {
 	return nil
 }
 
-// addVersionAnnotation adds the version annotation to nc.node
-func (nc *nodeConfig) addVersionAnnotation() {
-	nc.node.Annotations[VersionAnnotation] = version.Get()
-}
-
-// addPubKeyHashAnnotation adds the public key annotation to nc.node
-func (nc *nodeConfig) addPubKeyHashAnnotation() {
-	nc.node.Annotations[PubKeyHashAnnotation] = nc.publicKeyHash
-}
-
-// addAdditionalAnnotations merges nc.additionalAnnotations into the annotations on nc.node. If the annotation
-// already existed, its value will be overwritten.
-func (nc *nodeConfig) addAdditionalAnnotations() {
-	if nc.additionalAnnotations == nil {
-		return
-	}
-	for key, value := range nc.additionalAnnotations {
+// addAdditionalAnnotations adds the annotation with the given key and value, and updates the nodeconfig node object
+// If an annotation already exists, its value will be overwritten.
+func (nc *nodeConfig) addAnnotations(annotations map[string]string) error {
+	// Aggregate the string representation of a comma separated list of formatted annotations
+	patchData := "["
+	for key, value := range annotations {
+		// The `/` in the annotation key needs to be escaped in order to not be considered a "directory" in the path.
+		escapedAnnotation := strings.Replace(key, "/", "~1", -1)
+		patchData += fmt.Sprintf(`{"op":"add","path":"/metadata/annotations/%s","value":"%s"},`,
+			escapedAnnotation, value)
 		nc.node.Annotations[key] = value
 	}
+	// Replace last character (extra comma) with a closing brace
+	patchData = string(patchData[:len(patchData)-1]) + "]"
+	node, err := nc.k8sclientset.CoreV1().Nodes().Patch(context.TODO(), nc.node.GetName(), kubeTypes.JSONPatchType,
+		[]byte(patchData), meta.PatchOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "error updating annotations on node %s", nc.node.GetName())
```

Comment:
maybe add the patch data to the error here or add a log somewhere that helps us to debug this format?